### PR TITLE
Use distinct with strategies

### PR DESCRIPTION
**Pull request by bsideup**

### CODE REVIEW

File: core/src/main/java/org/testcontainers/dockerclient/DockerClientProviderStrategy.java [hunk 1]
```diff
@@ -137,10 +140,18 @@ public static DockerClientProviderStrategy getFirstValidStrategy(List<DockerClie
                                 .peek(strategy -> log.info("Loaded {} from ~/.testcontainers.properties, will try it first", strategy.getClass().getName())),
                         strategies
                                 .stream()
-                                .filter(it -> !it.getClass().getCanonicalName().equals(dockerClientStrategyClassName))
-                                .filter(DockerClientProviderStrategy::isApplicable)
                                 .sorted(Comparator.comparing(DockerClientProviderStrategy::getPriority).reversed())
                 )
+                .filter(new Predicate<DockerClientProviderStrategy>() {
```
Comment (kiview) [inline core/src/main/java/org/testcontainers/dockerclient/DockerClientProviderStrategy.java:145]: Why can't we use `distinct()` here?

Comment (bsideup) [reply core/src/main/java/org/testcontainers/dockerclient/DockerClientProviderStrategy.java:145]: It compares by value, while we need to compare by type

Comment (kiview) [reply core/src/main/java/org/testcontainers/dockerclient/DockerClientProviderStrategy.java:145]: D'oh, yes!

**Review by kiview — APPROVED** [review]

File: core/src/main/java/org/testcontainers/dockerclient/DockerClientProviderStrategy.java (hunk 1 above)
Comment (rnorth) [reply core/src/main/java/org/testcontainers/dockerclient/DockerClientProviderStrategy.java:145]: Streams API would benefit from a `distinct(Function<T,R>)` method, it seems 😂

**Review by rnorth — APPROVED** [review]

**Pull request merged**